CDP4Dal/Session.cs

```diff
@@ -64,9 +65,9 @@ public class Session : ISession
         private static Logger logger = LogManager.GetCurrentClassLogger();
 
         /// <summary>
-        /// The cancellation token source.
+        /// The cancellation token source dictionary.
         /// </summary>
-        private CancellationTokenSource cancellationTokenSource;
+        private readonly ConcurrentDictionary<Guid, CancellationTokenSource> cancellationTokenSourceDictionary;

```

Comment:
We might be able to use a HashSet here instead of a HashMap (or their C# equivalents) if the `CancellationTokenSource` objects are comparable.

Not critical though.

Reply:
there is no Concurrent hashset, so for me the use of the ConcurrentDictionary is fine